fvtest/compilertriltest/LogicalTest.cpp

```diff
@@ -59,14 +63,74 @@ TEST_P(Int32LogicalUnary, UsingConst) {
     ASSERT_EQ(exp, act);
 }
 
-
 INSTANTIATE_TEST_CASE_P(LogicalTest, Int32LogicalUnary, ::testing::Combine(
     ::testing::ValuesIn(TRTest::const_values<int32_t>()),
     ::testing::Values(
         std::make_tuple("ineg", ineg),
         std::make_tuple("iabs", iabs)
         )));
 
+class Int64LogicalUnary : public TRTest::UnaryOpTest<int64_t> {};
+
+TEST_P(Int64LogicalUnary, UsingConst) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
+    std::snprintf(inputTrees, 120,
+        "(method return=Int64
+           (block
+             (lreturn
+               (%s
+                 (lconst %d) )
+         )))",
+        param.opcode.c_str(),
+        param.value);
+
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int64_t (*)()>();
+    volatile auto exp = param.oracle(param.value);
+    volatile auto act = entry_point();
+    ASSERT_EQ(exp, act);
+}
+
+TEST_P(Int64LogicalUnary, UsingLoadParam) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
+    std::snprintf(inputTrees, 120,
+        "(method return=Int64 args=[Int64]
+           (block
+             (lreturn
+               (%s
+                 (lload parm=0))
+         )))",
+        param.opcode.c_str());
+
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int64_t (*)(int64_t)>();
+    ASSERT_EQ(param.oracle(param.lhs), entry_point(param.lhs));
```

Comment:
```suggestion
    ASSERT_EQ(param.oracle(param.value), entry_point(param.value));
```

Tests for unary opcodes don't have an `lhs`/`rhs`.